### PR TITLE
ROC-5811: use next working day for deadlines where appropriate

### PR DESCRIPTION
**Pull request by markry-kainos**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-5811

### Change description

https://github.com/hmcts/cmc-claim-store/pull/1025 needs to be merged first!

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/main/app/claims/models/claim.ts [hunk 1]
```diff
@@ -81,10 +82,11 @@ export class Claim {
   }
 
   get respondToMediationDeadline (): Moment {
-    if (!this.respondedAt) {
+    if (!this.mediationDeadline) {
       return undefined
     }
-    return this.respondedAt.clone().add('5', 'days')
+
```
Comment (andrewwa-kainos) [inline src/main/app/claims/models/claim.ts]: `if` condition is no longer needed

File: src/test/app/claims/calendarClient.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { MomentFactory } from 'shared/momentFactory'
+import { mockNextWorkingDay } from '../../http-mocks/claim-store'
```
Comment (andrewwa-kainos) [inline src/test/app/claims/calendarClient.ts]: Refactor relative imports please

File: src/main/app/claims/calendarClient.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { request } from 'client/request'
+import { claimApiBaseUrl } from 'claims/claimStoreClient'
+import { MomentFormatter } from 'utils/momentFormatter'
+import { Moment } from 'moment'
+import { MomentFactory } from 'shared/momentFactory'
+
+export class CalendarClient {
+
+  static getNextWorkingDay (date: Moment, addDays: number = 0): Promise<Moment> {
+    const url: string = `${claimApiBaseUrl}/calendar/next-working-day`
```
Comment (andrewwa-kainos) [inline src/main/app/claims/calendarClient.ts]: This doesn't need to be constructed each time the method is invoked; move `url` to the class itself

File: src/test/app/claims/calendarClient.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    const SATURDAY = MomentFactory.parse('2019-06-29')
+    const MONDAY_AFTER = '2019-07-01'
+    mockNextWorkingDay(MONDAY_AFTER)
+
+    const nextWorkingDay = await CalendarClient.getNextWorkingDay(SATURDAY, 0)
+
+    expect(nextWorkingDay.toISOString(false)).to.equal(MomentFactory.parse(MONDAY_AFTER).toISOString(false))
+  })
+
+})
```
Comment (andrewwa-kainos) [inline src/test/app/claims/calendarClient.ts]: Needs tests for `undefined` parameters, non-zero `addDays`, network failures, ...

File: src/test/app/claims/calendarClient.ts (hunk 2 above)
Comment (KiranVarma02) [reply src/test/app/claims/calendarClient.ts]: done

File: src/main/app/claims/models/claim.ts (hunk 1 above)
Comment (KiranVarma02) [reply src/main/app/claims/models/claim.ts]: done
